### DISCUSSION TITLE
magicbell-react throws 404 only the first time if the user doesn't exist

### DISCUSSION THREAD

**Discussion by cjstadther** [magicbell/community#68] [Questions] [answered]
We're using the react component, but the first time a user visits the page after logging in, they receive a 404 error from the component.

The reason looks to be that the first API call the component makes finds that there is no user with that external id, yet I'm pretty sure in the background it just created the user, because if the user refreshes their browser, they don't receive the error, and I can see a successful api call.  The other issue is that since the first API call failed, the websocket auth then fails (until the page is refreshed)

How can I handle this situation without directly adding the user to magicbell?  It would be great if the magicbell-react component could handle the first attempt by a new user without erroring.

Thanks

![magicbell1](https://user-images.githubusercontent.com/90266710/217100630-6a3c3364-cb0e-47bd-ac8a-f547e7aa307e.png)
![magicbell2](https://user-images.githubusercontent.com/90266710/217100648-f70d96d6-4e65-427a-bac1-d11e850132d1.png)

**Comment by unamashana** [comment, marked as answer]
Hey @cjstadther sorry about that. This usually happens if you init the component and then update the user props. Could that be happening in your code by any chance?

**Reply by cjstadther** [reply]
Thanks for responding.  The work-around I used was to make sure the users are created in magicbell when they are created in the system.  There was no code added I was using the straight magicbell-react component.  I was thinking it might have something to do with the HMAC for a user that doesn't exist yet.  
```
 <MagicBell
      apiKey={API_KEY}
      userExternalId={user_id}
      userKey={hmac}
      theme={theme}
      locale="en"
    >
      {props => <FloatingNotificationInbox width={400} height={500} {...props} />}
    </MagicBell>
```

**Reply by smeijer** [reply]
Hi @cjstadther, sorry for bringing this back to life. Do you by any chance know if any of the `apiKey`, `userExternalId ` and `userKey ` were `undefined` on first render?
